crates/core/tedge_api/src/entity_store.rs

```diff
@@ -492,11 +542,11 @@ impl EntityStore {
     /// If the provided fragment already existed, `false` is returned.
     pub fn update_twin_data(
         &mut self,
-        entity_topic_id: &EntityTopicId,
-        fragment_key: String,
-        fragment_value: JsonValue,
+        twin_message: EntityTwinMessage,
     ) -> Result<bool, entity_store::Error> {
```

Comment:
I failed to see the point of this `EntityTwinMessage` as they are built only to call the method `update_twin_data()`.